ctru-rs/src/error.rs

```diff
@@ -1,11 +1,46 @@
 use std::error;
 use std::ffi::CStr;
 use std::fmt;
+use std::ops::{ControlFlow, FromResidual, Try};
 
 use ctru_sys::result::{R_DESCRIPTION, R_LEVEL, R_MODULE, R_SUMMARY};
 
 pub type Result<T> = ::std::result::Result<T, Error>;
 
+#[derive(Debug, Clone, PartialEq, PartialOrd)]
+#[repr(transparent)]
+pub(crate) struct LibCtruResult(pub i32);
```

Comment:
@ian-h-chamberlain @AzureMarker could you give your opinion on the name? I find `LibCtruResult` a bit confusing (since there is already a `Result` type), but I can't come up with a much better idea...

Reply:
Maybe it would be a good idea to create a ``Result``-like enum, replace the current type-alias with that enum and implement: (i) the logic from ``LibCtruResult``, and (ii) a conversion method to Rust's ``Result``.

Though it feels like doing some gymnastics

Reply:
No, this new type is strictly internal to ease the raw `libctru` calls. It has no meaning esternally. Maybe `RawResultWrapper` or something else would be better… I am not very creative 😅

Reply:
Actually I like that name, definitely better that the current one!

Reply:
Should I change the current name, or should I wait for @ian-h-chamberlain and/or @AzureMarker opinions?

Reply:
You can change it already if you want, but this PR will stay on hold until I get some more opinions about it either way

Reply:
What do you think of `CtruSysResult` or `RawResult` instead? `RawResultWrapper` is a little long.

Reply:
`CtruSysResult` can be mistaken for `ctru_sys::Result`. `RawResult` is the better option imo.

Reply:
After reading through https://rust-lang.github.io/rfcs/3058-try-trait-v2.html#implementing-try-for-a-non-generic-type, `ResultCode` is another option.

Reply:
> I think `ResultCode` is the best option, to be honest

Yes, I would go with that.

Reply:
Changed to ``ResultCode`` (f896871b73f1b7d8dfa1e8c5c750637a4d913ca3)